modules/shared/src/main/scala/org/tessellation/schema/balance.scala

```diff
@@ -1,34 +1,51 @@
 package org.tessellation.schema
 
+import cats.kernel.Order
 import cats.syntax.either._
+import cats.syntax.option._
+import cats.syntax.semigroup._
 
 import scala.util.Try
 
 import derevo.cats.{eqv, show}
 import derevo.circe.magnolia.{decoder, encoder}
 import derevo.derive
+import eu.timepit.refined.cats._
 import eu.timepit.refined.numeric.NonNegative
 import eu.timepit.refined.refineV
-import eu.timepit.refined.types.numeric.NonNegBigInt
+import eu.timepit.refined.types.numeric.NonNegLong
 import io.estatico.newtype.macros.newtype
-import io.estatico.newtype.ops._
 import io.getquill.MappedEncoding
 
 object balance {
 
   @derive(decoder, encoder, eqv, show)
   @newtype
-  case class Balance(value: NonNegBigInt)
+  case class Balance(value: NonNegLong) {
+
+    def plus(that: Balance): Option[Balance] = {
```

Comment:
Is that correct that we add Balances? Shouldn't we add/subtract `Amount` to/from `Balance`?

Reply:
Then we need `Balance` -> `Amount` conversion. I guess it won't work the opposite way.

Reply:
I think so. It's like a semigroup on balance (but semigroup isn't partial so I cannot use semigroup itself).

Reply:
I can change it into `def plus(that: Amount)` but then I'd have to have conversion from `Balance` to `Amount` which seems a little bit weird 🤔 
IMO both ways are correct. wdyt?

Reply:
As agreed, moved to separate task

Reply:
I would build it around adding `Long` simply. Converting `Fee` or `Amount` to `Balance` just to perform adding doesn't look great. Anyway you are still getting `Option[Balance]` so you can pass `Long` and handle everything inside. Before adding or subtracting unwrap the type. That's what I think.

Reply:
We could also be fine with just one function then, like combine instead of two separate for adding and subtracting.